test/lib/cli/update-notifier.js

```diff
@@ -209,37 +231,48 @@ t.test('situations in which we do not notify', t => {
     const {
       wroteFile,
       result,
-      MANIFEST_REQUEST,
-    } = await runUpdateNotifier(t, { STAT_MTIME, version: HAVE_BETA })
+    } = await runUpdateNotifier(t, { STAT_MTIME, version: HAVE_BETA, PACOTE_MOCK_REQ_COUNT: 0 })
     t.equal(wroteFile, false, 'duration was not reset')
     t.equal(result, null)
-    t.strictSame(MANIFEST_REQUEST, [], 'no requests for manifests')
   })
 
   t.end()
 })
 
+t.test('notification situation with engine compatibility', async t => {
+  // no version which are greater than node 1.0.0 should be selected.
+  mockGlobals(t, { 'process.version': 'v1.0.0' }, { replace: true })
+
+  const {
+    wroteFile,
+    result,
+  } = await runUpdateNotifier(t, {
+    version: NEXT_VERSION_ENGINE_COMPATIBLE_MINOR,
+    PACOTE_MOCK_REQ_COUNT: 1 })
+
+  t.matchSnapshot(result)
+  t.equal(wroteFile, true)
+})
+
 t.test('notification situations', async t => {
   const cases = {
-    [HAVE_BETA]: [`^{V}`],
-    [NEXT_PATCH]: [`latest`, `^{V}`],
-    [NEXT_MINOR]: [`latest`, `^{V}`],
-    [CURRENT_PATCH]: ['latest'],
-    [CURRENT_MINOR]: ['latest'],
-    [CURRENT_MAJOR]: ['latest'],
+    [HAVE_BETA]: 1,
+    [NEXT_PATCH]: 2,
+    [NEXT_MINOR]: 2,
+    [CURRENT_PATCH]: 1,
+    [CURRENT_MINOR]: 1,
+    [CURRENT_MAJOR]: 1,
   }
 
-  for (const [version, reqs] of Object.entries(cases)) {
+  for (const [version, noOfRequests] of Object.entries(cases)) {
```

Comment:
```suggestion
  for (const [version, requestCount] of Object.entries(cases)) {
```

This matches the naming in `PACOTE_MOCK_REQ_COUNT`.